src/lib/userStorage/UserProfileStorage.js

```diff
@@ -0,0 +1,556 @@
+        break
+      default:
+        throw new Error('Invalid privacy setting', { privacy })
+    }
+    return display
+  }
+
+  /**
+   * Set profile field with privacy
+   * @param {*} field
+   * @param {*} value
+   * @param {*} privacy
+   * @param {*} onlyPrivacy
+   * @returns
+   */
+  setProfileField(
+    field: string,
+    value: string,
+    privacy: FieldPrivacy = 'public',
+    onlyPrivacy: boolean = false,
+  ): Promise<ACK> {
+    const display = this._setDisplayFieldBasedOnPrivacy(field, value, privacy)
+
+    logger.debug('setProfileField', { field, value, privacy, onlyPrivacy, display })
+
+    return this.setProfileFields({ [field]: { display, value, privacy } })
+  }
+
+  /**
+   * Avatar setter
+   * @returns {Promise<CID[]>}
+   */
+  async setAvatar(avatar) {
+    const cids = await this._resizeAndStoreAvatars(avatar)
+
+    await Promise.all(
+      // eslint-disable-next-line require-await
+      toPairs(cids).map(async ([field, value]) => this.setProfileField(field, value, 'public')),
+    )
+  }
+
+  /**
+   * Remove field from db
+   * @param field
+   * @returns {Promise<*>}
+   */
+  // eslint-disable-next-line require-await
+  async removeField(field: string) {
+    const profileToUpdate = this.profile
+    profileToUpdate[field] = { value: null, display: null, privacy: this.getFieldPrivacy(field) }
+    this._setLocalProfile(profileToUpdate)
+    return this.profiledb.removeField(field)
+  }
+
+  /**
+   * remove Avatar from profile
+   * @returns {Promise<[Promise<void>, Promise<void>, Promise<void>, Promise<void>, Promise<void>, Promise<void>, Promise<void>, Promise<void>, Promise<void>, Promise<void>]>}
+   */
+  async removeAvatar(): Promise<void> {
+    await Promise.all(
+      // eslint-disable-next-line require-await
+      ['avatar', 'smallAvatar'].map(async field =>
+        this.setProfileFields({ [field]: { value: null, display: null, privacy: this.getFieldPrivacy(field) } }),
+      ),
+    )
+  }
+
+  /**
+   * store Avatar
+   * @param avatarDataUrl
+   * @returns {Promise<{ avatar: CID, smallAvatar: CID }>}
+   */
+  async _resizeAndStoreAvatars(avatarDataUrl: string): Promise<{ avatar: string, smallAvatar: string }> {
+    let resizedDataUrl
+    const avatarSizes = [320, 50]
+
+    const resizedAvatars = await Promise.all(
+      avatarSizes.map(async size => {
+        resizedDataUrl = await resizeImage(resizedDataUrl || avatarDataUrl, size)
+
+        return resizedDataUrl
+      }),
+    )
+
+    // TODO: replace via IPFS.store() call once #3370 will be merged
+    const [avatar, smallAvatar] = await Promise.all(
+      // eslint-disable-next-line require-await
+      resizedAvatars.map(async dataUrl => Base64Storage.store(dataUrl)),
+    )
+
+    return { avatar, smallAvatar }
+  }
+
+  /**
+   * get another user public profile by wallet address
+   * @param {*} walletAddress
+   * @returns
+   */
+  getProfileByWalletAddress(walletAddress: string): Promise<any> {
```

Comment:
please check typings here too.
also re check are the methods defined as async really async
and add async qualifier where it's  missed